File: apps/admin_mobile/lib/domain/repositories/user_repository.dart
import '../entities/user.dart';
import '../../shared/types/paginated_result.dart';

abstract class UserRepository {
  Future<PaginatedResult<User>> getAll({UserRole? role, UserRole? excludeRole});
  Future<User> getById(int id);
  Future<User> invite(String email, UserRole role);
  Future<User> changeRole(int id, UserRole role);
}

File: apps/admin-v2/src/domain/repositories/user.repository.ts
import type { User, UserRole } from '../entities/user';
import type { PaginatedResult } from '../../shared/types/api';

export interface CreateMemberInput {
  name: string;
  username: string;
  password: string;
  email?: string | null;
  phone?: string | null;
  role: UserRole;
}

export interface UserRepository {
  getAll(filters?: { role?: UserRole; excludeRole?: UserRole }): Promise<PaginatedResult<User>>;
  getById(id: string): Promise<User>;
  invite(input: CreateMemberInput): Promise<User>;
  changeRole(id: string, role: UserRole): Promise<User>;
  resetPassword(id: string, password: string): Promise<void>;
}
